### DISCUSSION TITLE
About Android version of vtm

### DISCUSSION THREAD

**Discussion by USLTD** [directvt/vtm#202] [General]
Hello @o-sdn-o,
As far as I can see you don't maintain or build Android versions of vtm so I will build Android versions of releases (including demo which isn't available in Termux community). Thing is that I don't know how to release them. Should I do as PR (uploading built executables in archive or bare format) or edit Releases (but I don't have permission, except uploading in my fork).

**Comment by o-sdn-o** [comment]
Hi! Initially, I published the binaries, since it was impossible to build vtm using the compiler that came with the system. To do this, the user had to install a third-party compiler. Now the compilers that come with the OS support C++20 enough, and the user can build vtm himself without additional installation of exotic compilers.

Also, I'm not sure how useful it is to make binaries for operating systems that support a wide range of architectures. It will be necessary to build vtm for all possible combinations of OS/Architecture - this is a lot. 

I have statistics on downloading binaries for v0.6.0
```
"name": "vtm_demo_freebsd_amd64.tar.gz",  "download_count": 3,
"name": "vtm_demo_linux_amd64.tar.gz",    "download_count": 16,
"name": "vtm_demo_macos_any.tar.gz",      "download_count": 5,
"name": "vtm_demo_netbsd_amd64.tar.gz",   "download_count": 2,
"name": "vtm_demo_openbsd_amd64.tar.gz",  "download_count": 2,
"name": "vtm_demo_windows_amd64.zip",     "download_count": 10,

"name": "vtm_freebsd_amd64.tar.gz",       "download_count": 13,
"name": "vtm_linux_amd64.tar.gz",         "download_count": 222,
"name": "vtm_macos_any.tar.gz",           "download_count": 43,
"name": "vtm_netbsd_amd64.tar.gz",        "download_count": 4,
"name": "vtm_openbsd_amd64.tar.gz",       "download_count": 6,
"name": "vtm_windows_amd64.zip",          "download_count": 82,
```

Based on these statistics, demo binaries can already be excluded, as well as binaries for not very common systems. We can also think about how to simplify the building process. At the moment, the build is still complicated by the fact that Clang and GCC do not fully support C++20. Support for C++20 is constantly improving,  but still sometimes not enough, and on many systems it is still necessary to manually install the latest compiler in order to compile vtm.

I build for OpenBSD, FreeBSD and NetBSD mostly just to make sure vtm builds on a wide variety of OSes. I don't know how much it makes sense to do the same binaries for Android.

**Reply by USLTD** [reply]
Demo builds are awesome though... Although some of the demo apps are just proof of concepts, it is still nice to test them

**Comment by o-sdn-o** [comment]
If you have the ability to test the vtm build process for android, then that would be cool and that's enough. If you think that it is very non-trivial to build vtm on some common android-based systems, then it makes sense to publish such builds. It's like builds for MacOS, where it's not straight forward.

**Reply by USLTD** [reply]
I have ability to test vtm build process for Android although only aarch64 architecture (this is my Android device's architecture), but if it builds on aarch64 then it should build on other architectures too.
Well about builds I think only demo versions are needed since stable versions are automatically updated within few days of release.

**Reply by USLTD** [reply]
Also although outside of this topic but why don't you add automatic builds workflow?
https://github.com/marketplace/actions/automatic-releases

**Reply by USLTD** [reply]
It will automatically build on every PR/change

**Reply by o-sdn-o** [reply]
I don't use automatic builds workflow at the moment, because anyway I need to build vtm for various platforms and manually test and debug (this is one of the purposes of the demo mode besides proof of concept). So far, I do not know how to automate this testing process and cover it with tests, since problems appear in the most unexpected cases, just building it is not enough.
